ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java

```diff
@@ -236,37 +233,27 @@ private InputStream compressStream(InputStream uncompressedStream, boolean leave
 
     IngestionResult ingestFromBlob(BlobSourceInfo blobSourceInfo,
             IngestionProperties ingestionProperties,
-            BlobClient cloudBlockBlob,
             @Nullable String clientRequestId)
             throws IngestionClientException, IngestionServiceException {
         // trace ingestFromBlob
         return MonitoredActivity.invoke(
                 (SupplierTwoExceptions<IngestionResult, IngestionClientException, IngestionServiceException>) () -> ingestFromBlobImpl(blobSourceInfo,
-                        ingestionProperties, cloudBlockBlob, clientRequestId),
+                        ingestionProperties, clientRequestId),
                 getClientType().concat(".ingestFromBlob"),
                 getIngestionTraceAttributes(blobSourceInfo, ingestionProperties));
     }
 
-    private IngestionResult ingestFromBlobImpl(BlobSourceInfo blobSourceInfo, IngestionProperties ingestionProperties, BlobClient cloudBlockBlob,
+    private IngestionResult ingestFromBlobImpl(BlobSourceInfo blobSourceInfo, IngestionProperties ingestionProperties,
             @Nullable String clientRequestId)
             throws IngestionClientException, IngestionServiceException {
         String blobPath = blobSourceInfo.getBlobPath();
-        try {
-            // No need to check blob size if it was given to us that it's not empty
-            if (blobSourceInfo.getRawSizeInBytes() == 0 && cloudBlockBlob.getProperties().getBlobSize() == 0) {
```

Comment:
Do we check for empty blob elsewhere?

Reply:
what do you mean elsewhere ? here in java ? no